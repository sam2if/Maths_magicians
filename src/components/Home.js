import '../App.css';

export default function Home() {
  return (
    <div className="home-page">
      <h1
        style={{
          fontSize: '22px',
          fontWeight: '600',
          marginBottom: '2rem',
          lineHeight: '1em',
          color: '#000',
          textTransform: 'lowercase',
          textAlign: 'center',
          marginLeft: '-14rem',
        }}
      >
        Welcome to our page!
      </h1>
      <p>
        Lorem, ipsum dolor sit amet consectetur adipisicing elit. Sit ipsam quis
        ipsa repellat natus incidunt exercitationem nobis quo labore aspernatur.
        Consectetur voluptatibus consequatur veniam eaque deserunt, similique
        magnam libero dolorem!
      </p>
      <p>
        Lorem, ipsum dolor sit amet consectetur adipisicing elit. Sit ipsam quis
        ipsa repellat natus incidunt exercitationem nobis quo labore aspernatur.
        Consectetur voluptatibus consequatur veniam eaque deserunt, similique
        magnam libero dolorem!
      </p>
    </div>
  );
}
