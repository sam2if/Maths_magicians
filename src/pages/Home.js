/* eslint-disable */
import React from 'react';

const Home = () => (
  <div
    style={{
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      flexDirection: 'column',
      gap: '1rem',
      lineHeight: '1.5',
      padding: '2rem',
      color: '#F2EE9D',
      backgroundColor: '#7A9D54',
      opacity: '0.9',
      margin: '2rem 2rem',
    }}
  >
    <h1>Math Magicians</h1>
    <h3>Welcome to our page</h3>
    <p>
      Math magicians, also called mathemagicians, are people who combine math and magic to create impressive performances.
      They have discovered that math can be used to create tricks that seem impossible.
      By using their knowledge of math, they amaze audiences with tricks that defy logic.
    </p>
    <p>
      One interesting thing that math magicians have found is that math can be used to create mentalism tricks that work on their own.
      They have studied and experimented to uncover the hidden math patterns behind these tricks. By understanding and using these patterns.
      math magicians can predict things, read minds, and perform complex calculations that seem like magic.
    </p>
  </div>
);

export default Home;
