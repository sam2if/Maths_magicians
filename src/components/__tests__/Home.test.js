import { render, screen } from '@testing-library/react';
import Home from '../Home';

describe('Home component', () => {
  test('renders welcome message', () => {
    render(<Home />);
    const welcomeMessage = screen.getByText(/welcome to our page/i);
    expect(welcomeMessage).toBeInTheDocument();
  });

  test('applies correct styling to welcome message', () => {
    render(<Home />);
    const welcomeMessage = screen.getByText(/welcome to our page/i);
    expect(welcomeMessage).toHaveStyle({
      fontSize: '22px',
      fontWeight: '600',
      marginBottom: '2rem',
      lineHeight: '1em',
      color: '#000',
      textTransform: 'lowercase',
      textAlign: 'center',
      marginLeft: '-14rem',
    });
  });
});
