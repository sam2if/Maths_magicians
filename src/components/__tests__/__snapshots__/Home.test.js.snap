import { render, screen } from '@testing-library/react';
import Home from '../Home';

describe('Home component', () => {
  test('renders correctly', () => {
    const { container } = render(<Home />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
